torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -1467,7 +1459,7 @@ def _register_post_backward_reshard_only_hook(
     hook_handle = register_multi_grad_hook(
         inp_tensors, functools.partial(_post_backward_reshard, state, handle)
     )
-    flat_param._post_backward_hook_state = (hook_handle,)  # type: ignore[attr-defined, assignment]
+    handle.flat_param._post_backward_hook_handle = hook_handle # type: ignore[attr-defined, assignment]
```

Comment:
```suggestion
    flat_param._post_backward_hook_handle = hook_handle # type: ignore[attr-defined, assignment]
```
nit: this should be consistent with the above code.